### PR TITLE
[Kubernetes] Reference ECS fields and add agent.id field

### PR DESCRIPTION
**Pull request by constanca-m**
## What and why

Changes of this PR:

1. Reference ECS fields. This comes from [this comment](https://github.com/elastic/integrations/pull/8329#discussion_r1377259290) reference a similar issue https://github.com/elastic/integrations/pull/8100.
2. Add `agent.id` as dimension to every TSDB field, as it is one of the required fields ([reference](https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359)).

**Warning**: Running `elastic-package format` changed some of the files, but there are no differences in those apart from that.

Fields changed to ECS in `agent.yml`:
- cloud.account.id
- cloud.availability_zone
- cloud.instance.id
- cloud.instance.name
- cloud.machine.type
- cloud.provider
- cloud.region
- cloud.project.id
- host.architecture
- host.domain
- host.hostname
- host.id
- host.ip
- host.mac
- host.name
- host.os.family
- host.os.kernel
- host.os.name
- host.os.platform
- host.os.version
- host.type
- container.id
- container.image.name
- container.labels
- container.name

Fields changed to ECS in `base-fields.yml`:
- data_stream.type
- data_stream.dataset
- data_stream.namespace
- event.dataset
- '@timestamp'
- log.file.path - for logs data streams

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Results

Nothing should change.

I checked the dashboard of every data stream and they all work as before.

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
The purpose  of ecs.yml file in each data stream as I understand is to declare in it all the ecs fields. Does it make sense to have such a file and also have ecs fields declared in other files like base-fields.yml and agent.yml.
This happens also in other packages but I don't see why.

**Comment by constanca-m** [comment]
> Does it make sense to have such a file and also have ecs fields declared in other files like base-fields.yml and agent.yml.

I noticed it too, but I also don't know what was the logic followed to split all the fields between the files. Maybe moving them belongs to a different PR @MichaelKatsoulis

**Comment by MichaelKatsoulis** [comment]
> > Does it make sense to have such a file and also have ecs fields declared in other files like base-fields.yml and agent.yml.
> 
> I noticed it too, but I also don't know what was the logic followed to split all the fields between the files. Maybe moving them belongs to a different PR @MichaelKatsoulis

Yes. It seems that nobody is following the recommendations in https://github.com/elastic/integrations/blob/main/docs/fine_tune_integration.md (bullet 5).  
I believe: 
1. ecs.yml should contain all ecs fields of the data stream
2. agent.yml should contain all fields that are added by agent and are not ecs compliant (`host.containerized` for example)
3. base-fields.yml or package-fields.yml should contain all fields added by the package but not from the specific data stream and are not ecs compliant
4. fields.yml should contain all data stream specific fields

But yes this belongs to a different PR and honestly I don't know if it worths the trouble. @gizas , @axw what do you think ?

@constanca-m could you add in the description of this PR all the field names that you changed to ECS, so that we have them in one place?

**Review by MichaelKatsoulis — APPROVED** [review]

**Comment by axw** [comment]
> But yes this belongs to a different PR and honestly I don't know if it worths the trouble. @gizas , @axw what do you think ?

I don't know why some ECS fields are in `base-fields.yml` -- we also do that in the APM integration package. I'd probably just leave it, I don't think it really matters much.

File: packages/kubernetes/data_stream/apiserver/manifest.yml [hunk 1]
```diff
@@ -50,13 +50,13 @@ streams:
         show_user: false
         description: >
           Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the events are shipped. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
+
```
Comment (gizas) [inline packages/kubernetes/data_stream/apiserver/manifest.yml:53]: Minor you can remove this break line

File: packages/kubernetes/data_stream/audit_logs/fields/base-fields.yml [hunk 2]
```diff
@@ -1,25 +1,18 @@
...
 - name: log.offset
   type: long
   description: Offset of the entry in the log file.
 - name: log.file.path
-  type: keyword
-  description: Path to the log file.
+  external: ecs
 - name: input.type
   description: Type of input.
   type: keyword
```
Comment (gizas) [inline packages/kubernetes/data_stream/audit_logs/fields/base-fields.yml:18]: https://github.com/elastic/integrations/blob/89dd185408a5280b05e19e20ec7fc3127576b2a5/packages/kubernetes/data_stream/audit_logs/fields/ecs.yml dont we need agent.id in ecs.yml here?

Comment (gizas) [reply packages/kubernetes/data_stream/audit_logs/fields/base-fields.yml:18]: Is only needed for TSDB and becasue here were are referring to logs we dont need it right?

**Review by gizas — APPROVED** [review]

File: packages/kubernetes/data_stream/audit_logs/fields/base-fields.yml (hunk 2 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/audit_logs/fields/base-fields.yml:18]: > Is only needed for TSDB and becasue here were are referring to logs we dont need it right? 

Yes, only for TSDB data streams, and logs data streams are never TSDB.

File: packages/kubernetes/data_stream/apiserver/manifest.yml (hunk 1 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/apiserver/manifest.yml:53]: It was added by `elastic-agent format`.

**Pull request merged**